src/Avatar/BehaviorPipeline.cs

```diff
@@ -76,12 +76,12 @@ public BehaviorPipeline()
         /// <returns>Return value from the pipeline.</returns>
         public IMethodReturn Invoke(IMethodInvocation invocation, bool throwOnException = false)
         {
-            IMethodReturn CallBaseOrThrow(IMethodInvocation invocation) => invocation.SupportsCallBase ?
-                invocation.CreateCallBaseReturn() :
-                throw new NotImplementedException(ThisAssembly.Strings.PipelineNotImplemented(invocation));
+            IMethodReturn InvokeTargetOrThrow(IMethodInvocation invocation) => invocation.HasImplementation ?
+                invocation.CreateInvokeReturn() :
+                throw new NotImplementedException(ThisAssembly.Strings.NotImplemented(invocation));
 
             if (Behaviors.Count == 0)
-                return CallBaseOrThrow(invocation);
+                return InvokeTargetOrThrow(invocation);
```

Comment:
`InvokeTarget...` or `InvokeImplementation...` (or `InvokeImplementationOnTarget...`)? A few more comments about the same thing will follow below.

Reply:
InvokeImplementation +1